ccdproc/bitmask.py

```diff
@@ -0,0 +1,473 @@
+            If ``False`` then the ``bitpos`` will be interpreted as active
+            bitpos. To reverse the meaning so that ``bitpos`` indicate which
+            bitpos to ignore set this to ``True``.
+            Default is ``False``.
+
+        Returns
+        -------
+        mask : `numpy.ndarray` of boolean dtype
+            The mask created from the bitmask.
+
+        Notes
+        -----
+        If all ``flags`` should be taken into account then consider using
+        ``bitmask.data.astype(bool)``.
+
+        Duplicate items in the ``bitpos``, ``bitval`` and ``bitname`` are
+        ignored by the function.
+
+        Examples
+        --------
+        The active flags can be given by their bit position (from right to
+        left)::
+
+            >>> from ccdproc import Bitmask
+            >>> bm = Bitmask(np.arange(8))
+            >>> bm.to_mask(bitpos=[1])
+            array([False, False,  True,  True, False, False,  True,  True], dtype=bool)
+
+        or by their bit value (``position ** 2``)::
+
+            >>> bm.to_mask(bitval=[2])
+            array([False, False,  True,  True, False, False,  True,  True], dtype=bool)
+
+        or if a ``name_map`` is given also by their name::
+
+            >>> bm.name_map['bad'] = 1
+            >>> bm.to_mask(bitname=['bad'])
+            array([False, False,  True,  True, False, False,  True,  True], dtype=bool)
+
+        It is also possible to specify only the **inactive** flags::
+
+            >>> bm.to_mask(bitname=['bad'], reverse_flag_interpretation=True)
+            array([False,  True, False,  True,  True,  True,  True,  True], dtype=bool)
+
+        and to specify multiple flags::
+
+            >>> bm.to_mask(bitpos=[0, 1])
+            array([False,  True,  True,  True, False,  True,  True,  True], dtype=bool)
+        """
+        bitmask = self.data
+        # Make sure one and only one of the three options to provide the
+        # "active" flags is actually given.
+        if sum(inp is not None for inp in (bitpos, bitval, bitname)) != 1:
+            raise TypeError('One and only one of "bitpos", "bitval" or '
+                            '"bitname" may be not "None".')
+
+        # So that the logic and validation is only applied once process
+        # first "bitname" to "bitpos" and then "bitpos" to "bitval". So each
+        # step is quite short and well defined and the logic can be implemented
+        # only in the "bitval" branch.
+
+        # Map the names to bit positions
+        if bitname is not None:
+            try:
+                bitpos = [self.name_map[name] for name in bitname]
+            except KeyError:
+                raise ValueError(
+                    'The "name(s)" {} were not found in "self.name_map" ({})'
+                    .format([name for name in bitname
+                             if name not in self.name_map],
+                            self.name_map))
+            bitname = None
```

Comment:
``bitname = None`` not necessary. I think you can do ``elif`` for the next ``if``.